src/App.tsx

```diff
@@ -1,24 +1,160 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect, useMemo } from 'react';
 import { UserWarning } from './UserWarning';
+import { TodoHeader } from './components/TodoHeader/TodoHeader';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoFilter } from './components/TodoFilter/TodoFilter';
+import { Notification } from './components/Notification/Notification';
 
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import { TodoStatus } from './types/TodoStatus';
+import { TodoErrors } from './types/TodoErrors';
+import {
+  getTodos, USER_ID, createTodo, deleteTodo,
+} from './api/todos';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<TodoErrors | null>(null);
+  const [status, setStatus] = useState<TodoStatus>(TodoStatus.All);
+  const [title, setTitle] = useState<string>('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isProcessed, setIsProcessed] = useState<boolean>(false);
+  const [deletedId, setDeletedId] = useState<number | null>(null);
+
+  const deleteTodoHandler = (todoId: number) => {
+    setDeletedId(todoId);
+
+    deleteTodo(todoId)
+      .then(() => {
+        setTodos(filtredTodos => filtredTodos
+          .filter(todo => todo.id !== todoId));
+      })
+      .catch(() => setError(TodoErrors.Delete));
+  };
+
+  const createTodoHandler = (event: React.FormEvent<HTMLFormElement>) => {
```

Comment:
Missing useCallback hook for the createTodoHandler function, which could cause unnecessary re-renders.